networkit/cpp/io/NetworkitBinaryWriter.cpp

```diff
@@ -210,6 +319,141 @@ void NetworkitBinaryWriter::write(const Graph &G, const std::string& path) {
 			});
 		}
 	});
+	// Write adj weights.
+	for (uint64_t c = 1; c < chunks; c++) {
+		outfile.write(reinterpret_cast<char*>(&adjWghtOffsets[c-1]), sizeof(uint64_t));
+	}
+	G.forNodes([&](node u) {
+		uint8_t tmp [10];
+		G.forNeighborsOf(u,[&](node v,edgeweight w){
+			uint64_t weightSize;
+			if(!G.isDirected()) {
+				if(v <= u) {
+					switch(weightFormat) {
+						case nkbg::WEIGHT_FORMAT::unsignedFormat:
+							weightSize = encode(w, tmp);
+							outfile.write(reinterpret_cast<char*>(tmp), weightSize);
+						break;
+						case nkbg::WEIGHT_FORMAT::doubleFormat:
+							{
+								outfile.write(reinterpret_cast<char*>(&w), sizeof(double));
+							}
+						break;
+						case nkbg::WEIGHT_FORMAT::signedFormat:
+							{
+								uint64_t weight = encodeZigzag(w);
+								weightSize = encode(weight,tmp);
+								outfile.write(reinterpret_cast<char*>(tmp), weightSize);
+							}
+						break;
+						case nkbg::WEIGHT_FORMAT::floatFormat:
+							{
+								float weight = w;
+								outfile.write(reinterpret_cast<char*>(&weight), sizeof(float));
+							}
+						break;
+					}
+				}
+			} else {
+				switch(weightFormat) {
+					case nkbg::WEIGHT_FORMAT::unsignedFormat:
+						{
+							weightSize = encode(w, tmp);
+							outfile.write(reinterpret_cast<char*>(tmp), weightSize);
+						}
+					break;
+					case nkbg::WEIGHT_FORMAT::doubleFormat:
+						{
+							outfile.write(reinterpret_cast<char*>(&w), sizeof(double));
+						}
+					break;
+					case nkbg::WEIGHT_FORMAT::signedFormat:
+						{
+							int64_t weight = encodeZigzag(w);
+							weightSize = encode(weight,tmp);
+							outfile.write(reinterpret_cast<char*>(tmp), weightSize);
+						}
+					break;
+					case nkbg::WEIGHT_FORMAT::floatFormat:
+						{
+							float weight = w;
+							outfile.write(reinterpret_cast<char*>(&weight), sizeof(float));
+						}
+					break;
+				}
```

Comment:
This code is duplicated 4 times, right? Could we move it into a lambda function?